mypy/stubtest.py

```diff
@@ -204,7 +209,7 @@ def verify_mypyfile(
     to_check = set(
         m
         for m, o in stub.names.items()
-        if not o.module_hidden and (not m.startswith("_") or hasattr(runtime, m))
+        if not o.module_hidden and (not is_probably_private(m) or hasattr(runtime, m))
```

Comment:
I was planning something very similar, which was why I factored it out into a helper function even though it was only used once in my patch 😆